hazelcast/src/main/java/com/hazelcast/internal/nearcache/impl/store/AbstractNearCacheRecordStore.java

```diff
@@ -275,6 +275,7 @@ public V get(K key) {
                 }
                 if (staleReadDetector.isStaleRead(key, record)) {
                     remove(key);
+                    nearCacheStats.incrementMisses();
```

Comment:
the main reason to add it was aware of removes because of detected staleness. Expirations have its own stats updated that we can be aware of them. But i see your reasoning lets put  it in another PR